File: apps/client-macos/Sources/TimeTrack/Projects/Project.swift
import Foundation

/// Client-side mirror of `ProjectSchema` / `TaskSchema` in @timetrack/contracts.
/// Named `ProjectTask` because `Task` is Swift's concurrency primitive (CLAUDE.md /
/// plan Global Constraints).
struct Project: Codable, Identifiable, Equatable {
    let id: String
    let teamId: String
    let name: String
    let archived: Bool
    let tasks: [ProjectTask]?
}

struct ProjectTask: Codable, Identifiable, Equatable {
    let id: String
    let projectId: String
    let name: String
}

File: apps/client-windows/src/NiftyTimer/Projects/Project.cs
using System.Text.Json.Serialization;

namespace NiftyTimer.Projects;

/// <summary>Client-side mirror of <c>ProjectSchema</c> in @timetrack/contracts.</summary>
public sealed record Project(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("teamId")] string TeamId,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("archived")] bool Archived,
    [property: JsonPropertyName("tasks")] IReadOnlyList<ProjectTask>? Tasks);

/// <summary>Client-side mirror of <c>TaskSchema</c> in @timetrack/contracts.</summary>
public sealed record ProjectTask(
    [property: JsonPropertyName("id")] string Id,
    [property: JsonPropertyName("projectId")] string ProjectId,
    [property: JsonPropertyName("name")] string Name);

/// <summary>The picker selection worth remembering across launches. Ids only — never names.</summary>
public sealed record StoredSelection(
    [property: JsonPropertyName("projectId")] string ProjectId,
    [property: JsonPropertyName("taskId")] string? TaskId);
